File: frontend/lib/models/customer.dart
// lib/models/customer.dart

class Customer {
  final int      id;
  final String   name;
  final String   phone;   
  final String?  email;    // ← nullable
  final String?  gst;      // ← nullable
  final String?  address;  // ← nullable
  final String? state;

  Customer({
    required this.id,
    required this.name,
    required this.phone,
    this.email,             // optional
    this.gst,               // optional
    this.address,           // optional
    this.state
  });

  factory Customer.fromJson(Map<String, dynamic> json) => Customer(
        id:      json['id'] as int,
        name:    json['name'] as String,
        phone:   json['phone'] as String,
        email:   json['email'] as String?,       // may be null
        gst:     json['gst'] as String?,  // matches backend field
        address: json['address'] as String?,     // may be null
        state:   json['state'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'name':       name,
        'phone':      phone,
        'email':      email,
        'gst':        gst,
        'address':    address,
        'state':      state,
      };
}

File: backend/models/customer.py
# heymachi_backend/models/customer.py

from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from .base import Base

class Customer(Base):
    __tablename__ = "customers"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    phone = Column(String, nullable=False, unique=True)
    email = Column(String, nullable=True)
    gst = Column(String, nullable=True)
    address = Column(String, nullable=True)
    state = Column(String, nullable=True)
    business_id = Column(String, ForeignKey("business_accounts.id"))
    
    business = relationship("BusinessAccount", backref="customers")
    bills = relationship("Bill", back_populates="customer")
